src/components/RetreatButton.js

```diff
@@ -1,23 +1,35 @@
 import React from "react";
-import styled from "styled-components";
+import styled, { css } from "styled-components";
 import Pointer from "../ressources/icons/Pointer";
 
 const RetreatStyled = styled.button`
-  background: ${props => props.theme.contrast};
-  color: ${props => props.theme.brightFont};
-  font-size: 24px;
-  padding-left: 35px;
-  padding-right: 10px;
-  width: 260px;
-  height: 55px;
-  display: flex;
-  align-items: center;
-  justify-content: space-between;
+  ${props =>
+    props.disabled
+      ? css`
+          display: none;
+        `
+      : css`
+          background: ${props.theme.contrast};
+          color: ${props.theme.brightFont};
+          font-size: 24px;
+          padding-left: 35px;
+          padding-right: 10px;
+          width: 260px;
+          height: 55px;
+          display: flex;
+          align-items: center;
+          justify-content: space-between;
+          border: none;
+        `}
 `;
 
-function RetreatButton({ name }) {
+function RetreatButton({ handleClick, name, inGame }) {
   return (
-    <RetreatStyled name={name}>
+    <RetreatStyled
```

Comment:
Another way to write this is not via display none but to only generate the element when disabled is false.

${ disabled && ( <RetreatStyled disabled={inGame})}

only when disabled is true RetreatStyled will be rendered.

Reply:
Thx, mate. This is a better solution for this usecase! I will modify the solution and implement it!